watertap/costing/watertap_costing_package.py

```diff
@@ -400,10 +400,44 @@ def build_global_params(self):
             var.fix()
 
         # Define standard material flows and costs
-        self.defined_flows["electricity"] = self.electricity_base_cost
-        self.defined_flows["NaOCl"] = self.naocl_cost / self.naocl_purity
-        self.defined_flows["CaOH2"] = self.caoh2_cost / self.caoh2_purity
-        self.defined_flows["steam"] = self.steam_unit_cost
+        # The WaterTAP costing package creates flows
+        # in a lazy fashion, the first time `cost_flow`
+        # is called for a flow. The `available_flows`
+        # are all the flows which can be costed with
+        # the WaterTAP costing package.
+        self.available_flows = {}
+        self.available_flows["electricity"] = self.electricity_base_cost
+        self.available_flows["NaOCl"] = self.naocl_cost / self.naocl_purity
+        self.available_flows["CaOH2"] = self.caoh2_cost / self.caoh2_purity
+        self.available_flows["steam"] = self.steam_unit_cost
+
+    def cost_flow(self, flow_expr, flow_type):

```

Comment:
Do we have unit tests that cover this feature?

Reply:
I have now added one in 33d086ef.